src/main/java/it/unibz/inf/ontouml/vp/listeners/DiagramListener.java

```diff
@@ -44,6 +45,8 @@ public void diagramElementAdded(IDiagramUIModel diagram, IDiagramElement shape)
 			System.out.println("Exception caught while adding element to diagram.");
 			e.printStackTrace();
 		}
+		
+		SmartProperties.smartMetaProperties();
```

Comment:
Should we trigger `SmartProperties.smartMetaProperties()` on the listeners or on the `ApplyStereotype.java`?

Reply:
If we do it on the listeners, won't this be called too many times? I mean, if the user drags and drop the elements in the diagram, the listener is called again.

Reply:
Consider the following scenario:

1. the user creates association `x` between classes `A` and `B`
2. the user applies the /</<material>> to `x`
3. the tool automatically sets the cardinalities on `x`
4. the user deletes these cardinalities
5. the user reassigns `x` to relate the classes `A` and `C`
6. the tool adds the multiplicities again